src/withJWTProps.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import wrapDisplayName from 'recompose/wrapDisplayName'
+import jwtDecode from 'jwt-decode'
+
+export default (propName, propsMapping) => Target => {
+  class WithJWTProps extends Component {
+    constructor(props, context) {
+      super(props, context)
+
+      this.state = this.buildState(props[propName])
+    }
+
+    componentWillReceiveProps(nextProps) {
+      if (nextProps[propName] !== this.props[propName]) {
+        this.setState(this.buildState(nextProps[propName]))
+      }
+    }
+
+    buildState(token) {
+      try {
+        const decodedToken = jwtDecode(token)
```

Comment:
Fair enough. As long as no consuming project needs to verify the signature of the token, then it will be fine. When someone does, though, they will need to use a different jwt library, which will mean a breaking change to this.